marlowe-playground-client/grammar.ne

```diff
@@ -0,0 +1,180 @@
+@{%
+const moo = require("moo");
+
+const lexer = moo.compile({
+        WS: /[ \t]+/,
+        number: /0|-?[1-9][0-9]*/,
+        string: {match: /"(?:\\["\\]|[^\n"\\])*"/, value: x => x.slice(1, -1)},
+        comma: ',',
+        lparen: '(',
+        rparen: ')',
+        lsquare: '[',
+        rsquare: ']',
+        hole: /\?[a-zA-Z0-9_-]+/,
+        CONSTRUCTORS: {
+            match: /[A-Z][A-Za-z]+/, type: moo.keywords({
```

Comment:
This probably won't make any difference, but I think the `+` in this regular expression should be a `*`, because there can be single character constructors.

Reply:
Actually, I just noticed in the deployment that if you use a single character constructor it doesn't give an error. It may be because of this combined with something else.
EDIT: it seems errors with lexer are shown very subtly, so you can only tell because all buttons are frozen

Reply:
hmm, there should still be an error marker there? Can you give an example. FYI I am going to make an error pane in the new design.

Reply:
Just the contract `C` for example. If you want a counter example for after replacing the `+` with `*` you can just add an `ñ` anywhere in the middle of white spaces of a valid contract.

Reply:
Ah yes I see, I know what I need to do and it needs some work on this PR.

Reply:
ok, pushed, proper errors with lexer errors now.

Reply:
Just deployed, can you check